src/contrib/cluster/Akka.DistributedData/Replicator.cs

```diff
@@ -695,7 +695,11 @@ private DataEnvelope Write(string key, DataEnvelope writeEnvelope)
             var envelope = GetData(key);
             if (envelope != null)
             {
-                if (ReferenceEquals(envelope, writeEnvelope)) return writeEnvelope;
+                if (envelope.Equals(writeEnvelope))
```

Comment:
Used this to fix bug related to duplicate `Update` events for identical operations in the `ReplicatorSpec` - this needed to be equality by value, not by reference.